sdk-extensions/autoconfigure/src/test/java/io/opentelemetry/sdk/autoconfigure/LoggerProviderConfigurationTest.java

```diff
@@ -43,10 +46,11 @@ void configureLoggerProvider() {
         DefaultConfigProperties.createForTest(properties),
         LoggerProviderConfiguration.class.getClassLoader(),
         MeterProvider.noop(),
-        (a, unused) -> a);
-    SdkLoggerProvider loggerProvider = builder.build();
+        (a, unused) -> a,
+        closeables);
+    cleanup.addCloseables(closeables);
 
-    try {
+    try (SdkLoggerProvider loggerProvider = builder.build()) {
```

Comment:
This type of change is a common theme in this PR. Switching to try-with-resources when available to cleanup the code, and also block (for at most 10 seconds) until the resource is closed.